filebeat/autodiscover/builder/hints/config.go

```diff
@@ -39,3 +39,25 @@ func defaultConfig() config {
 		Config: cfg,
 	}
 }
+
+// Unpack is needed here as go-ucfg fails to unpack the Config object by default
+func (c *config) Unpack(from *common.Config) error {
+	tmpConfig := struct {
+		Key string `config:"key"`
+	}{
+		Key: c.Key,
+	}
+	if err := from.Unpack(&tmpConfig); err != nil {
+		return err
+	}
+
+	if config, err := from.Child("config", -1); err == nil {
```

Comment:
we do not need to unpack the config here and can be:
```
	if config, err := from.Child("config", -1); err == nil {
		c.Config = config
	}

```
This would help builders that inherit this one like how we do.